ledger/simulation/simulator.go

```diff
@@ -189,9 +196,14 @@ func (s Simulator) simulateWithTracer(txgroup []transactions.SignedTxn, tracer l
 }
 
 // Simulate simulates a transaction group using the simulator. Will error if the transaction group is not well-formed.
-func (s Simulator) Simulate(txgroup []transactions.SignedTxn) (Result, error) {
-	simulatorTracer := makeEvalTracer(s.ledger.start, txgroup)
-	block, missingSigIndexes, err := s.simulateWithTracer(txgroup, simulatorTracer)
+func (s Simulator) Simulate(simulateRequest Request) (Result, error) {
+	simulatorTracer := makeEvalTracer(s.ledger.start, simulateRequest)
+
+	if len(simulateRequest.TxnGroups) != 1 {
```

Comment:
nit: now that we have this check here, you could remove the one from line 1011 of `daemon/algod/api/server/v2/handlers.go`

Reply:
see [updated eval test, repeat 40 times line 1050 length, eval-changes to …](https://github.com/algorand/go-algorand/pull/5247/commits/04436a3ebb3923b05ccc7a0614f7d4bc7254e057)